Dockerfile

```diff
@@ -1,10 +1,41 @@
-FROM elixir:1.10-alpine
-RUN mix local.hex --force
-RUN mix local.rebar --force
+# ---- Build Stage ----
+FROM elixir:1.10-alpine AS builder
+
+# Set environment variables for building the application
+ENV MIX_ENV=prod \
+    TEST=1 \
+    LANG=C.UTF-8
+
+# Install hex and rebar
+RUN mix local.hex --force && \
+    mix local.rebar --force
+
+
+# Create the application build directory
 WORKDIR /app
-ADD . /app
+
+# Copy over all the necessary application files and directories
+COPY config ./config
+COPY lib ./lib
+COPY priv ./priv
+COPY mix.exs .
+COPY mix.lock .
+
+# Fetch the application dependencies and build the application
 RUN mix deps.get
 RUN mix deps.compile
-RUN mix compile
-CMD mix phx.server
-EXPOSE 4000
+RUN mix release
+
+# ---- Application Stage ----
+FROM alpine AS runtime
+
+ENV LANG=C.UTF-8
+
+# Install openssl
+RUN apk update && apk add openssl ncurses-libs
```

Comment:
do we really need ncurses?

Reply:
yep, it's needed by `erts-10.7.1/bin/beam.smp`